optimade/filtertransformers/mongo.py

```diff
@@ -172,30 +172,84 @@ def property_zip_addon(self, arg):
 
     def _recursive_expression_phrase(self, arg):
         """Helper function for parsing `expression_phrase`. Recursively sorts out
-        the correct precedence for `$and`, `$or` and `$nor`.
+        the correct precedence for `$not', '$and` and `$or`.
 
         """
-        if len(arg) == 1:
-            # without NOT
-            return arg[0]
 
-        # handle the case of {"$not": {"$or": [expr1, expr2]}} using {"$nor": [expr1, expr2]}.
-        if "$or" in arg[1]:
-            return {"$nor": self._recursive_expression_phrase([arg[1]["$or"]])}
+        def handle_not_and(arg):
+            # Handle the case of {"$not": {"$and": [expr1, expr2]}}
+            # which is equal to {"or": [{"$not": expr1}, {"$not": expr2}]}}
+            # We have to check for the special case in which the "and" was created by a previous NOT
+            # e.g NOT (NOT ({"a": {"$eq" : 6}})) -> NOT({$and:[{"a": {"$ne" : 6}},{"a": {"$ne": None}}]})
+
+            expr1 = arg["$and"][0]
+            expr2 = arg["$and"][1]
+            key = list(expr1.keys())[0]
```

Comment:
What happens in the case of `NOT(A AND B AND C)`? i.e. more than 2 expressions?

Reply:
I have added a test for this to test_mongo.py and the code handles this case correctly. 
`{"a": {"$ne": None}}` is always created as a pair with another clause with `{"a": {...}}`, so it is not a problem that we only test the first two clauses. 
The order is also always the same so the check for `expr1.get(key) == {"$ne": None}` is not strictly necessary, although it does allow reordering of the arguments in a future version.

I have also added a test to tests.server.query_params.test_filter. To test the behaviour of NOT(A AND B) which was the case that failed in the original issue.